core/state/statedb.go

```diff
@@ -884,16 +849,6 @@ func (s *StateDB) IntermediateRoot(deleteEmptyObjects bool) common.Hash {
 			s.prefetcher = nil
 		}()
 	}
-	// Although naively it makes sense to retrieve the account trie and then do
-	// the contract storage and account updates sequentially, that short circuits
-	// the account prefetcher. Instead, let's process all the storage updates
-	// first, giving the account prefetches just a few more milliseconds of time
-	// to pull useful data from disk.
```

Comment:
You kind of elegantly undid this optimization with the prefetcher :P

Reply:
Cc @rjl493456442

Reply:
myeah... I got the point...

I am fine to revert the change